src/MinterGateway.sol

```diff
@@ -1066,58 +1074,88 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
-        if (threshold_ > 0) {
-            uint256 requiredThreshold_ = updateCollateralValidatorThreshold();
+        uint256 requiredThreshold_ = updateCollateralValidatorThreshold();
 
-            unchecked {
-                // NOTE: By this point, it is already established that `threshold_` is less than `requiredThreshold_`.
-                revert NotEnoughValidSignatures(requiredThreshold_ - threshold_, requiredThreshold_);
-            }
-        }
+        if (validCount_ < requiredThreshold_) revert NotEnoughValidSignatures(validCount_, requiredThreshold_);
+    }
+
+    /**
+     * @dev    Checks that a signature is a valid validator signature.
+     * @param  minter_       The address of the minter.
+     * @param  collateral_   The amount of collateral.
+     * @param  retrievalIds_ The list of outstanding collateral retrieval IDs to resolve.
+     * @param  metadataHash_ The hash of metadata of the collateral update, reserved for future informational use.
+     * @param  validator_    The address of a validator.
+     * @param  timestamp_    The timestamp for the collateral update signature.
+     * @param  signature_    The signature from the validator.
+     * @return valid_        Whether the signature is a valid validator signature.
+     */
+    function _verifyValidatorSignature(
+        address minter_,
+        uint256 collateral_,
+        uint256[] calldata retrievalIds_,
+        bytes32 metadataHash_,
+        address validator_,
+        uint256 timestamp_,
+        bytes calldata signature_
+    ) internal returns (bool valid_) {
+        // Check that the timestamp is not 0.
+        // NOTE: Revert here because this failure is entirely within the minter's control.
+        if (timestamp_ == 0) revert ZeroTimestamp();
+
+        // Check that the timestamp is not in the future.
+        // NOTE: Revert here because this failure is entirely within the minter's control.
+        if (timestamp_ > uint40(block.timestamp)) revert FutureTimestamp();
+
+        uint256 lastTimestamp_ = _lastSignatureTimestamp[minter_][validator_];
+
+        // Check that the timestamp is not older than the last signature timestamp.
+        // NOTE: Revert here because this failure is entirely within the minter's control.
+        if (timestamp_ <= lastTimestamp_) revert OutdatedValidatorTimestamp(validator_, timestamp_, lastTimestamp_);
+
+        // Check that validator is approved by TTG.
+        if (!isValidatorApprovedByTTG(validator_)) return false;
+
+        // Check that ECDSA or ERC1271 signatures for given digest are valid.
+        valid_ = SignatureChecker.isValidSignature(
+            validator_,
+            _getUpdateCollateralDigest(minter_, collateral_, retrievalIds_, metadataHash_, timestamp_),
+            signature_
+        );
+
+        // Save the last signature timestamp for the minter and validator combination.
+        _lastSignatureTimestamp[minter_][validator_] = timestamp_;
```

Comment:
```suggestion
        if (!isValidatorApprovedByTTG(validator_)) {
            valid_ = false;
        } else {
            // Check that ECDSA or ERC1271 signatures for given digest are valid.
            valid_ = SignatureChecker.isValidSignature(
                validator_,
                _getUpdateCollateralDigest(minter_, collateral_, retrievalIds_, metadataHash_, timestamp_),
                signature_
            );

            // Save the last signature timestamp for the minter and validator combination.
            _lastSignatureTimestamp[minter_][validator_] = timestamp_;
        }
```

Reply:
having one code block is easier for the reader of code vs if {} else {}, but it made me realize that this code can be unsafe

Reply:
` _lastSignatureTimestamp[minter_][validator_] = timestamp_;` should be done only for valid signatures 
@deluca-mike @PierrickGT 
more thought is needed here

Reply:
> having one code block is easier for the reader of code vs if {} else {}, but it made me realize that this code can be unsafe

`valid_` being a named returned variable, it should be assigned and not implicitly returned.
Or if we want to return, we should also do it for the else case.

Reply:
```suggestion
        if (!isValidatorApprovedByTTG(validator_)) return false;

        // Check that ECDSA or ERC1271 signatures for given digest are valid.
        bool isValid_ = SignatureChecker.isValidSignature(
            validator_,
            _getUpdateCollateralDigest(minter_, collateral_, retrievalIds_, metadataHash_, timestamp_),
            signature_
        );

        if (isValid_) {
            // Save the last signature timestamp for the minter and validator combination.
            _lastSignatureTimestamp[minter_][validator_] = timestamp_;
        }

        return isValid_;
```